paimon-core/src/main/java/org/apache/paimon/mergetree/compact/FirstRowMergeFunction.java

```diff
@@ -0,0 +1,88 @@
+/*
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.apache.paimon.mergetree.compact;
+
+import org.apache.paimon.KeyValue;
+import org.apache.paimon.data.serializer.InternalRowSerializer;
+import org.apache.paimon.types.RowKind;
+import org.apache.paimon.types.RowType;
+import org.apache.paimon.utils.Preconditions;
+
+import javax.annotation.Nullable;
+
+/**
+ * A {@link MergeFunction} where key is primary key (unique) and value is the full record, only keep
+ * the first one.
+ */
+public class FirstRowMergeFunction implements MergeFunction<KeyValue> {
+
+    private final InternalRowSerializer keySerializer;
+    private final InternalRowSerializer valueSerializer;
+    private KeyValue first;
+
+    protected FirstRowMergeFunction(RowType keyType, RowType valueType) {
+        this.keySerializer = new InternalRowSerializer(keyType);
+        this.valueSerializer = new InternalRowSerializer(valueType);
+    }
+
+    @Override
+    public void reset() {
+        this.first = null;
+    }
+
+    @Override
+    public void add(KeyValue kv) {
+        RowKind rowKind = kv.valueKind();
+        Preconditions.checkArgument(
+                rowKind.isAdd(),
+                String.format("First row merge engine don't accept %s message", rowKind));
```

Comment:
Do not use `String.format`, you should use:
```
checkArgument(
            boolean condition,
            @Nullable String errorMessageTemplate,
            @Nullable Object... errorMessageArgs
```
This will not invoke `String.format` in normal code path.